plugin/hover.py

```diff
@@ -82,17 +83,16 @@ def run(
         point: Optional[int] = None,
         event: Optional[dict] = None
     ) -> None:
-        temp_point = point
-        if temp_point is None:
+        point_or_region = point # type: Optional[PointOrRegion]
+        if point_or_region is None:
             region = first_selection_region(self.view)
             if region is not None:
-                temp_point = region.begin()
-        if temp_point is None:
+                point_or_region = region
+        if point_or_region is None:
             return
         window = self.view.window()
         if not window:
             return
-        hover_point = temp_point
```

Comment:
Actually changes here seem unnecessary.

We don't need to pass range to `request_symbol_hover_async`. It can figure out what to do itself based on the hover point.

If your intention was to let `lsp_hover` be passed a range then I don't see the point of that due to above.

Reply:
Well, almost, if not the fact that we want to get relevant diagnostics.

Reply:
But then you are not even using `diagnostics_intersecting_region_async` so it doesn't work as I would expect anyway...

Reply:
I removed the usage of `diagnostics_intersecting_region_async` when it is a region because one test is failing and I couldn't figure out what was happening to even fix it. Maybe the test case was wrong ?

Reply:
Test server doesn't support range request so it should be unaffected by these changes if the changes are done correctly. It's probably related to this issue since those changes seem to change the behavior in some cases.